rd-kt/rd-framework/src/main/kotlin/com/jetbrains/rd/framework/base/RdExtBase.kt

```diff
@@ -51,8 +51,11 @@ abstract class RdExtBase : RdReactiveBase() {
         lifetime.bracket(
             {
 //                extScheduler = sc
-                extProtocol = Protocol(parentProtocol.name, parentProtocol.serializers, parentProtocol.identity, sc, extWire, lifetime, serializationContext, parentProtocol.contexts).also {
+                extProtocol = Protocol(parentProtocol.name, parentProtocol.serializers, parentProtocol.identity, sc, extWire, lifetime, serializationContext, parentProtocol.contexts, parentProtocol.extCreatedLocal, createExtSignal()).also {
                     it.outOfSyncModels.flowInto(lifetime, super.protocol.outOfSyncModels) { model -> model }
+                    parentProtocol.extCreatedNetworked.localChange {
```

Comment:
Am I right that the code won't be executed for the case of local ext creation since the `lifetime` will be terminated?
And that is the guarantee behind that  
```
extCreatedNetworked.advise(lifetime) { message ->
  if (extCreatedNetworked.isLocalChange) return@advise
    extCreatedLocal.fire(message)
}
 ```
 
 If that's the case, I find it very confusing and hard to understand. It hides the real reasons by the peculiarities and corner cases of lifetime-related API that skips the execution for `Lifetime.Terminated`, while leaving out the code that uses local cookies for remote updates with unclear execution flow. It's hard to tell for sure if a particular API won't throw an exception in the case of `Lifetime.Terminated`, so usually people don't mess with terminated lifetimes. 
 
Sorry if that's not the case

Reply:
This is a tricky way to use ext's delayed message sending. There is only one local signal, passed through all protocols, and a remote signal for each protocol instance (including ext protocols). It's quite complex, indeed, but has nothing to do with Lifetime API.